### DISCUSSION TITLE
What is the purpose of Correlation PreviousRequestId in logs?

### DISCUSSION THREAD

**Discussion by ahmedtolba1984** [ThreeMammals/Ocelot#1865] [Q&A] [answered]
Dears,
I already ready documentation on CorrelationIDs**/**Tracing but I can't understand the purpose of PreviousRequestId in log files

**Comment by raman-m** [comment, marked as answer]
### Documentation
- [Request ID](https://ocelot.readthedocs.io/en/latest/features/requestid.html)
- [Logging](https://ocelot.readthedocs.io/en/latest/features/logging.html)

### Answer
- [PreviousRequestId represents ID of the previous request](https://ocelot.readthedocs.io/en/latest/features/logging.html#request-id:~:text=PreviousRequestId%20represents%20ID%20of%20the%20previous%20request)

But after our review of the Logging design during release [22.0.0](https://github.com/ThreeMammals/Ocelot/releases/tag/22.0.0), as a team we would say, this **PreviousRequestId** sub-feature is broken. We see empty values in logs always.
So, we keep in mind this problem.

**Reply by raman-m** [reply]
@ggnaegi  FYI
Do you see empty values of this property in logs of the production?

**Reply by ggnaegi** [reply]
> @ggnaegi  FYI
> Do you see empty values of this property in logs of the production?

@raman-m for a normal request you will never have a non-null previous id (http context is a scoped service). Maybe for request aggregation?

**Reply by raman-m** [reply]
@ggnaegi  Can you prove that attaching logs from prod please? Much appreciate that.
I'm a bit confused cause I thought this feature is broken. But you said the feature is functioning. I need to look at real logs...
